### PR TITLE
fixes #2669 username first name and last name doesn't accept i18n charac...

### PR DESCRIPTION
**Pull request by abenari**
...ters

### CODE REVIEW

**Comment by domcleal** [comment]
Are you sure the username constraints should be loosened too?  Maybe check with @ares on Signo's constraints.

**Comment by ohadlevy** [comment]
when i tried with hebrew chars it didnt actually work, it showed ???? instead of אוהד... not sure why.
any change to the user afterward was denied.

**Comment by ares** [comment]
From Signo perspective, username is used to construct OpenID identifier, and only requirement for this is that it can be used as URI. Username is in path fragment of URI. If there are some encoding issues, we should use %xx mechanism (see [1]). Also we should avoid reserved characters, see section 3.3 in [1] for path fragment.

[1] http://www.ietf.org/rfc/rfc2396.txt)

File: app/models/user.rb [hunk 1]
```diff
@@ -44,9 +44,9 @@ class User < ActiveRecord::Base
   validates_presence_of :login, :auth_source_id
   validates_presence_of :password_hash, :if => Proc.new {|user| user.manage_password?}
   validates_confirmation_of :password,  :if => Proc.new {|user| user.manage_password?}, :unless => Proc.new {|user| user.password.empty?}
-  validates_format_of :login, :with => /^[a-z0-9_\-@\.]*$/i
+  validates_format_of :login, :with => /^[^<|>|\/|\s]*$/i
```
Comment (pitr-ch) [inline app/models/user.rb]: I think in ruby 1.9 `/[[:alnum:]_-]/` will match all UTF8 chars, see http://rubular.com/r/6MRH5fg91M

**Comment by abenari** [comment]
[test]

**Comment by domcleal** [comment]
I can't get this to work, is it just me?

I've tried a first name of "Ãdmin" and also the test from the BZ, "Гесер".  Both pass the rubular test, but I keep getting validation errors.

(Ruby 2.0.0 btw)

**Comment by abenari** [comment]
should apply PR #708 first to test this.

**Comment by GregSutcliffe** [comment]
@abenari @domcleal I also can't make this work - I've applied both 708 and 723 and it still won't let me save a new user. Error is:

Failed to save: Surname is invalid

Ruby 1.8.7 (just for comparison)

**Comment by domcleal** [comment]
I've tested again with #708 but no luck, I'm unable to edit my own user and give the first name "Ãdmin".  If I comment out the validation line, it does save, so I guess alnum isn't enough?  Does it depend on my own locale perhaps?

**Comment by GregSutcliffe** [comment]
I should have noticed the fact that 1.8 was supposed to be not working - 1.9 appears to work correctly. Maybe a ruby 2.0 problem @domcleal ?

**Comment by domcleal** [comment]
Indeed, it works under 1.9.3 and not 2.0.0.  I wonder if that's deliberate, or a bug?

**Comment by GregSutcliffe** [comment]
Ruby 2.0.0-p195 works for me @domcleal

**Comment by domcleal** [comment]
Ok, given I'm on 2.0.0-p0 I'm going to assume it's a bug.  Merged as aa925f8, thanks @abenari.

**Pull request closed without merging**